Added test cases for JSON resource marshaling

sayalideshmukh10 opened the pull request:
Added test cases for the below structs - github/scim.go (file)

1. UpdateAttributeForSCIMUserOperations
2. UpdateAttributeForSCIMUserOptions
3. ListSCIMProvisionedIdentitiesOptions

gmlewis commented on github/scim_test.go:
```diff
@@ -389,3 +389,59 @@ func TestSCIMUserAttributes_Marshal(t *testing.T) {
+	}
+
+	want := `{
+		"op":"TestOp",
```
```suggestion
		"op": "TestOp",
```

gmlewis commented on github/scim_test.go:
```diff
@@ -389,3 +389,59 @@ func TestSCIMUserAttributes_Marshal(t *testing.T) {
+	}
+
+	want := `{
+		"schemas":["test", "schema"],
```
```suggestion
		"schemas": ["test", "schema"],
```

gmlewis commented on github/scim_test.go:
```diff
@@ -389,3 +389,59 @@ func TestSCIMUserAttributes_Marshal(t *testing.T) {
+	want := `{
+		"schemas":["test", "schema"],
+		"operations":{
+			"op":"TestOp",
```
```suggestion
			"op": "TestOp",
```

gmlewis commented on github/scim_test.go:
```diff
@@ -389,3 +389,59 @@ func TestSCIMUserAttributes_Marshal(t *testing.T) {
+
+	want := `{
+		"schemas":["test", "schema"],
+		"operations":{
```
```suggestion
		"operations": {
```

gmlewis commented on github/scim_test.go:
```diff
@@ -389,3 +389,59 @@ func TestSCIMUserAttributes_Marshal(t *testing.T) {
+	}
+
+	want := `{
+		"startIndex":1,
```
```suggestion
		"startIndex": 1,
```

gmlewis commented on github/scim_test.go:
```diff
@@ -389,3 +389,59 @@ func TestSCIMUserAttributes_Marshal(t *testing.T) {
+	want := `{
+		"startIndex":1,
+		"count": 10,
+	 	"filter":"test"
```
```suggestion
	 	"filter": "test"
```

gmlewis reviewed (changes requested):
Thank you, @sayalideshmukh10 .
We might as well make the formatting of the wanted JSON consistent and easy-to-read, as if it were created by a JSON indented marshaler.

sayalideshmukh10 commented:
Hello @gmlewis,
I addressed all review comments 

I am a first-time contributor to Hackoctober, It will be a great help if you can accept my contribution.

gmlewis reviewed (approved):
Thank you, @sayalideshmukh10 !
Before we merge, would you like to resolve the conflicts with the recent merges to the master branch, or would you like me to do it?

sayalideshmukh10 commented:
Thank you @gmlewis for asking about merging conflicts and resolving 
I already resolved the merge conflicts kindly look into it
